ext/fg/js/source.js

```diff
@@ -79,7 +82,17 @@ class TextSourceRange {
     }
 
     equals(other) {
-        return other && other.range && other.range.compareBoundaryPoints(Range.START_TO_START, this.range) === 0;
+        if (other === null) {
+            return false;
+        }
+        if (this.imposterSourceElement !== null) {
+            return (
+                this.imposterSourceElement === other.imposterSourceElement &&
+                this.rangeStartOffset === other.rangeStartOffset
+            );
+        } else {
+            return this.range.compareBoundaryPoints(Range.START_TO_START, other.range) === 0;
```

Comment:
Is the comparison swapped here on purpose? I think `other` is the newer `TextSourceRange` here and `this.range` might have already lost its element. I don't know how `compareBoundaryPoints` behaves when called on a range with a removed element vs when it's given one as an argument.

Reply:
I would hope that there's no difference between the two, besides one being the negative of the other. I swapped it because it more closely matches the order of the this/other ops of the `equals` function.

https://dom.spec.whatwg.org/#dom-range-compareboundarypoints

Reply:
I agree this looks more logical, but I was thinking the decision to do it the other way around could have been due to a bug in the past. Anyway I tested it and it worked fine so I'll approve it like this. Just wanted to hear your point of view.